rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,169 @@
+from rotkehlchen.typing import KusamaAddress
+from rotkehlchen.user_messages import MessagesAggregator
+
+from .typing import KusamaNodeName, SubstrateChain, SubstrateChainData, SubstrateNode
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+"""
+TODO priority 1:
+- Check whether we need to set up the RuntimeConfiguration and if so, how to
+set up multiple RuntimeConfiguration. Otherwise check the statement that keeps
+the decodes up to date:
+  * https://github.com/polkascan/py-substrate-interface#keeping-type-registry-presets-up-to-date
+  * Fix `ExtrinsicMetadata` error
+  * https://github.com/polkascan/py-scale-codec/blob/3db9ac54ae467c0c653a85c9224bce7ba38f778f/scalecodec/base.py#L381
+  * https://github.com/polkascan/py-scale-codec#multiple-runtime-configurations
+- Manually test cases:
+  - Add/edit/delete KSM account/accounts
+  - Validate balances and totals (use accounts without KSM as well)
+- Improve node connection (check how EthereumManager does):
+  - Smoke request for testing node connectivity?
+  - Add weight to to nodes, etc.
+- Exception handling (e.g RemoteError, SubstrateRequestException)
+- ChainManager attribute, how do we want to use/access the instances?
+  1. self.substrate.kusama, and self.substrate.polkadot:
+    - Single argument/property "substrate"
+  2. self.kusama and self.polkadot
+    - One argument/property per chain.
+- Polish init attributes
+- Improve logging, use msg_aggregator
+- For SubstrateChain enum consider to use the values of SupportedBlockchain:
+  * 'KSM' instead of 1
+- Typings / naming:
+  * SubstrateNode (from SubtrateInterface instantiation)
+- Run unit tests (expected to have broken ones).
+
+TODO priority 2:
+- Issue 1920: add ability to use a local Kusama node
+    * https://github.com/rotki/rotki/issues/1920
+    * Either include it in KusamaNodeName or consider to use a generic structure
+    for any substrate local node.
+- Implement tests
+
+TODO priority 3:
+- Find/ask for a contract that allows batch queries for balances
+"""
+class SubstrateManager():
+    def __init__(
+            self,
+            chain: Literal['kusama'],
+            msg_aggregator: MessagesAggregator,
+    ) -> None:
+        self.supported_chains = {str(chain_) for chain_ in SubstrateChain}
+        if chain not in self.supported_chains:
+            raise AttributeError(f'Unexpected SubstrateManager chain: {chain}')
+
+        self.chain = chain
+        self.chain_data = self._get_substrate_chain_data()
+        # self._load_type_registry_preset()
+        self.substrate = self._initialize_substrate_interface()
+        log.debug(
+            f'Initializing Substrate Manager for {chain} with {self.chain_data.node_name} node',
+        )
+        self.msg_aggregator = msg_aggregator
+
+    def _get_substrate_chain_data(self) -> SubstrateChainData:
+        if self.chain == str(SubstrateChain.KUSAMA):
+            node_name = KusamaNodeName.PARITY
```

Comment:
This has been sorted with:
- `SubstrateInterfaceAttributes`: this namedtuple stores the essential attributes for instantiating `SubstrateInterface` for the given chain. It may seem overkilling have it in a namedtuple for a single property (only `type_registry_preset` at the moment), but `SubstrateInterface` is quite configurable in terms of constructor attributes so I prefer this way.
- `ChainProperties`: once we instantiate `SubstrateInterface` for the given chain per node, we do our first request. This first request pulls [the chain properties](https://github.com/polkascan/py-substrate-interface/blob/master/substrateinterface/base.py#L566) (you can check it is a request to the chain) and populates this structure.
- `request_available_nodes()` this function used as a decorator calls the wrapped function per each available node. The sequence of nodes is set by `_get_nodes_call_order()`. At the moment, with the absence of "own node" and our criteria (like the EthereumManager `OPEN_NODES_WEIGHT_MAP`), the preference is determined by their synchronisation with the chain; the higher the block, the higher the preference. I've tested it and works.